conda/base/context.py

```diff
@@ -447,12 +447,15 @@ def custom_channels(self):
 
     @property
     def channels(self):
-        # add 'defaults' channel when necessary if --channel is given via the command line
+        # add 'defaults' channel if 'channels' is not defined in configuration files
+        # or environment variables, and 'defaults' is not already given with a `--channel` CLI flag
         if self._argparse_args and 'channel' in self._argparse_args:
             # TODO: it's args.channel right now, not channels
             argparse_channels = tuple(self._argparse_args['channel'] or ())
-            if argparse_channels and argparse_channels == self._channels:
-                return argparse_channels + (DEFAULTS_CHANNEL_NAME,)
+            if not any('channels' in params and location != 'cmd'
+                       for location, params in iteritems(self.raw_data)):
```

Comment:
This would evaluate to False if a channel is defined in an environment variable but no entry is present in the .condarc file.  I believe `defaults` should be included in this case.

Reply:
Let me double check

Reply:
Yup, you were right.  It didn't matter here, because the argparse argument is named `channel`, not `channels`.  I was hoping to change that at the conda 4.4 transition, although I'm not sure that actually got in.

Either way, if it ever does someday, there are now regression tests that should have us covered :)

Reply:
For whatever reason, I didn't see this comment when I made those previous comments.

At least for what my understanding was, and what I outlined in https://github.com/conda/conda/issues/6228, I had intended to do the magic of automatically adding `defaults` ONLY if no other configuration for channels was supplied.  That included environment variables.

Let's rethink that though.  At least discuss it.  Thinking through it, I still come down on the side of not magically adding `defaults` if ANY other configuration is set for channels.  I think there are two reasons.

1. If a user explicitly sets an environment variable, they probably know enough about what they're doing that they'll include `defaults` if they want it.
2. It becomes much more inconvenient from the CLI perspective to get to the negative case of *not* adding the `defaults` channel.  What I mean, is that, if I really, really don't want `defaults`, but only for this operation, right now I have an option that doesn't force me to modify a file and then modify it back after my operation is complete.  Say I only want conda-forge for the one operation.

        conda config --add channels conda-forge
        conda config --remove channels defaults
        conda install some-package
        conda config --remove channels conda-forge

    or, edit condarc directly, but edit it twice, before and after.

Those are my thoughts anyway.  What are yours?